manticore/ethereum.py

```diff
@@ -1141,6 +1145,11 @@ def flagged(flag):
                 summary.write("\n")
 
 
+            if blockchain._sha3.items():
+                summary.write("Known hashes:\n")
+            for key,value in blockchain._sha3.items():
+                summary.write('%s::%x\n'%(key.encode('hex'), value))
+
```

Comment:
This will generate a redundant `items` list. Suggestion, either:

```
sha3s = blockchain._sha3.items()
if sha3s:
    summary.write("Known hashes:\n")
    for key, value in sha3s:
        summary.write('%s::%x\n'%(key.encode('hex'), value))
```

or

```
if blockchain._sha3:
    summary.write("Known hashes:\n")
    for key, value in blockchain._sha3.items():
        ...
```

I think I like the last option in terms of readability.